flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/declarative/StateTrackingMockExecutionGraph.java

```diff
@@ -0,0 +1,459 @@
+    public JobID getJobID() {
+        return jobId;
+    }
+
+    @Override
+    public String getJobName() {
+        return "testJob";
+    }
+
+    @Nullable
+    @Override
+    public ErrorInfo getFailureInfo() {
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public ArchivedExecutionConfig getArchivedExecutionConfig() {
+        return archivedExecutionConfig;
+    }
+
+    @Override
+    public boolean isStoppable() {
+        return false;
+    }
+
+    @Nullable
+    @Override
+    public CheckpointCoordinatorConfiguration getCheckpointCoordinatorConfiguration() {
+        return null;
+    }
+
+    @Nullable
+    @Override
+    public CheckpointStatsSnapshot getCheckpointStatsSnapshot() {
+        return null;
+    }
+
+    @Override
+    public Optional<String> getStateBackendName() {
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<String> getCheckpointStorageName() {
+        return Optional.empty();
+    }
+
+    @Override
+    public StringifiedAccumulatorResult[] getAccumulatorResultsStringified() {
+        return new StringifiedAccumulatorResult[0];
+    }
+
+    // -- remaining interface implementations: all unsupported
+
+    @Override
+    public boolean isArchived() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void start(@Nonnull ComponentMainThreadExecutor jobMasterMainThreadExecutor) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getNumberOfExecutionJobVertices() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public SchedulingTopology getSchedulingTopology() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public ScheduleMode getScheduleMode() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Nonnull
+    @Override
+    public ComponentMainThreadExecutor getJobMasterMainThreadExecutor() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void enableCheckpointing(
+            CheckpointCoordinatorConfiguration chkConfig,
+            List<MasterTriggerRestoreHook<?>> masterHooks,
+            CheckpointIDCounter checkpointIDCounter,
+            CompletedCheckpointStore checkpointStore,
+            StateBackend checkpointStateBackend,
+            CheckpointStorage checkpointStorage,
+            CheckpointStatsTracker statsTracker,
+            CheckpointsCleaner checkpointsCleaner) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Nullable
+    @Override
+    public CheckpointCoordinator getCheckpointCoordinator() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public KvStateLocationRegistry getKvStateLocationRegistry() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void setJsonPlan(String jsonPlan) {
+        throw new UnsupportedOperationException();
```

Comment:
We could implement this in a simple way

Reply:
Yes, but why should I spend time on something I don't need?